cmd/program/program.go

```diff
@@ -172,7 +171,21 @@ func (p *Project) CreateMainFile() error {
 		return err
 	}
 
-	return nil
+	// Initialize git repo
+    err = utils.ExecuteCmd("git", []string{"init"}, projectPath)
+    if err != nil {
+        log.Printf("Error initializing git repo: %v", err)
+        cobra.CheckErr(err)
+        return err
+    }
+    // Create gitignore
+    err = utils.ExecuteCmd("npx", []string{"gitignore", "go"}, projectPath)
+    if err != nil {
+        log.Printf("Error creating gitignore: %v", err)
+        cobra.CheckErr(err)
+        return err
+    }
+    return nil
```

Comment:
What if the user does not have NPX installed? 
I am wondering if it makes more sense to follow the exciting template pattern with a .gitignore file that we can create and then inject?

Reply:
yeah npx could be a problem. i will use existing template pattern to inject .gitignore file

Reply:
the npx gitignore command simply downloaded the chosen gitignore file from this repo https://github.com/github/gitignore

This can be done in pure Go without NPX